app/Model/BuildErrorFilter.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/*=========================================================================
+  Program:   CDash - Cross-Platform Dashboard System
+  Module:    $Id$
+  Language:  PHP
+  Date:      $Date$
+  Version:   $Revision$
+
+  Copyright (c) Kitware, Inc. All rights reserved.
+  See LICENSE or http://www.cdash.org/licensing/ for details.
+
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE. See the above copyright notices for more information.
+=========================================================================*/
+namespace CDash\Model;
+
+use CDash\Database;
+use PDO;
+
+class BuildErrorFilter
+{
+    private $WarningsFilter;
+    private $ErrorsFilter;
+    private $ProjectId;
+
+    public function __construct($projectid)
+    {
+        $this->WarningsFilter = null;
+        $this->ErrorsFilter = null;
+        $this->ProjectId = $projectid;
+
+        $this->PDO = Database::getInstance()->getPdo();
+    }
+
+    public function FilterWarning($warning)
+    {
+        if (is_null($this->WarningsFilter)) {
+            $stmt = $this->PDO->prepare(
+                'SELECT warnings FROM build_filters WHERE projectid = ?');
+            pdo_execute($stmt, [$this->ProjectId]);
+            $this->WarningsFilter = $stmt->fetchColumn();
+        }
+
+        if ($this->WarningsFilter) {
+            return preg_match($this->WarningsFilter, $warning) === 1;
```

Comment:
If you have multiple warnings that you'd like to surpress it would be challenging to write a single regular expression that matches them all.

I agree that `projectid` makes the most sense for a primary key on your new `build_filter` table.  How about changing  the `errors` and `warnings` column on this table to be `text` fields and storing each such regular expression as a separate line?

Reply:
This is a good idea.